homeassistant/config_entries.py

```diff
@@ -517,6 +517,15 @@ async def async_setup(
 
         # Only store setup result as state if it was not forwarded.
         if domain_is_integration := self.domain == integration.domain:
+            if self.state in (
+                ConfigEntryState.LOADED,
+                ConfigEntryState.SETUP_IN_PROGRESS,
+            ):
+                raise OperationNotAllowed(
+                    f"The config entry {self.title} ({self.domain}) with entry_id"
+                    f" {self.entry_id} cannot be setup because is already loaded in the"
```

Comment:
nit:
```suggestion
                    f" {self.entry_id} cannot be set up because is already loaded in the"
```

Taking another look, maybe also:
```suggestion
                    f" {self.entry_id} cannot be set up because it is already loaded in the"
```

Reply:
I'll do a followup as I copied the message from the other location

Reply:
I'm adding a lock check to async_setup and async_unload to ensure we never run them without the lock in https://github.com/home-assistant/core/pull/117214.  I fixed the text there